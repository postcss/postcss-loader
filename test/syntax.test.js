import {
  compile,
  getCompiler,
  getErrors,
  getCodeFromBundle,
  getWarnings,
} from './helpers';

describe('"syntax" option', () => {
  it('should work with "String" value', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        syntax: 'sugarss',
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.sss', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with "Object" value', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        // eslint-disable-next-line global-require
        syntax: require('sugarss'),
      },
    });

    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.sss', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should throw an error on "unresolved" syntax', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        syntax: 'unresolved',
      },
    });
    const stats = await compile(compiler);

    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats, true)).toMatchSnapshot('errors');
  });
});
