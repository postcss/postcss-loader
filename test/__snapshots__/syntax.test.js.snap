// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"syntax" option should throw an error on "unresolved" syntax: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
TypeError: Cannot read property 'parse' of undefined",
  "ModuleError: Module Error (from \`replaced original path\`):
Loading PostCSS \\"unresolved\\" syntax failed: Cannot find module 'unresolved' from 'src/utils.js'",
]
`;

exports[`"syntax" option should throw an error on "unresolved" syntax: warnings 1`] = `Array []`;

exports[`"syntax" option should work with "Object" value: css 1`] = `
"a
  color: black
"
`;

exports[`"syntax" option should work with "Object" value: errors 1`] = `Array []`;

exports[`"syntax" option should work with "Object" value: warnings 1`] = `Array []`;

exports[`"syntax" option should work with "String" value: css 1`] = `
"a
  color: black
"
`;

exports[`"syntax" option should work with "String" value: errors 1`] = `Array []`;

exports[`"syntax" option should work with "String" value: warnings 1`] = `Array []`;
